File: src/main/java/org/apache/fineract/client/services/MappingFinancialActivitiesToAccountsApi.java
/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements. See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership. The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
package org.apache.fineract.client.services;

import org.apache.fineract.client.CollectionFormats.*;

import rx.Observable;
import retrofit2.http.*;

import okhttp3.RequestBody;
import okhttp3.ResponseBody;
import okhttp3.MultipartBody;

import org.apache.fineract.client.models.DeleteFinancialActivityAccountsResponse;
import org.apache.fineract.client.models.GetFinancialActivityAccountsResponse;
import org.apache.fineract.client.models.PostFinancialActivityAccountsRequest;
import org.apache.fineract.client.models.PostFinancialActivityAccountsResponse;
import org.apache.fineract.client.models.PutFinancialActivityAccountsResponse;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public interface MappingFinancialActivitiesToAccountsApi {
  /**
   * Create a new Financial Activity to Accounts Mapping
   * Mandatory Fields financialActivityId, glAccountId
   * @param postFinancialActivityAccountsRequest  (optional)
   * @return Observable&lt;PostFinancialActivityAccountsResponse&gt;
   */
  @Headers({
    "Content-Type:application/json"
  })
  @POST("financialactivityaccounts")
  Observable<PostFinancialActivityAccountsResponse> createGLAccount(
    @retrofit2.http.Body PostFinancialActivityAccountsRequest postFinancialActivityAccountsRequest
  );

  /**
   * Delete a Financial Activity to Account Mapping
   * 
   * @param mappingId mappingId (required)
   * @return Observable&lt;DeleteFinancialActivityAccountsResponse&gt;
   */
  @DELETE("financialactivityaccounts/{mappingId}")
  Observable<DeleteFinancialActivityAccountsResponse> deleteGLAccount(
    @retrofit2.http.Path("mappingId") Long mappingId
  );

  /**
   * Retrieve a Financial Activity to Account Mapping 
   * Example Requests:  financialactivityaccounts/1
   * @param mappingId mappingId (required)
   * @return Observable&lt;GetFinancialActivityAccountsResponse&gt;
   */
  @GET("financialactivityaccounts/{mappingId}")
  Observable<GetFinancialActivityAccountsResponse> retreive(
    @retrofit2.http.Path("mappingId") Long mappingId
  );

  /**
   * List Financial Activities to Accounts Mappings
   * Example Requests:  financialactivityaccounts
   * @return Observable&lt;List&lt;GetFinancialActivityAccountsResponse&gt;&gt;
   */
  @GET("financialactivityaccounts")
  Observable<List<GetFinancialActivityAccountsResponse>> retrieveAll();
    

  /**
   * 
   * 
   * @return Observable&lt;String&gt;
   */
  @GET("financialactivityaccounts/template")
  Observable<String> retrieveTemplate();
    

  /**
   * Update a Financial Activity to Account Mapping
   * the API updates the Ledger account linked to a Financial Activity  
   * @param mappingId mappingId (required)
   * @param postFinancialActivityAccountsRequest  (optional)
   * @return Observable&lt;PutFinancialActivityAccountsResponse&gt;
   */
  @Headers({
    "Content-Type:application/json"
  })
  @PUT("financialactivityaccounts/{mappingId}")
  Observable<PutFinancialActivityAccountsResponse> updateGLAccount(
    @retrofit2.http.Path("mappingId") Long mappingId, @retrofit2.http.Body PostFinancialActivityAccountsRequest postFinancialActivityAccountsRequest
  );

}

File: src/main/kotlin/org/openapitools/client/apis/MappingFinancialActivitiesToAccountsApi.kt
package org.openapitools.client.apis

import org.openapitools.client.models.DeleteFinancialActivityAccountsResponse
import org.openapitools.client.models.GetFinancialActivityAccountsResponse
import org.openapitools.client.models.PostFinancialActivityAccountsRequest
import org.openapitools.client.models.PostFinancialActivityAccountsResponse
import org.openapitools.client.models.PutFinancialActivityAccountsResponse
import retrofit2.http.Body
import retrofit2.http.DELETE
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.PUT
import retrofit2.http.Path

interface MappingFinancialActivitiesToAccountsApi {
    /**
     * Create a new Financial Activity to Accounts Mapping
     * Mandatory Fields financialActivityId, glAccountId
     * Responses:
     *  - 200: OK
     *
     * @param postFinancialActivityAccountsRequest  (optional)
     * @return [PostFinancialActivityAccountsResponse]
     */
    @POST("v1/financialactivityaccounts")
    suspend fun createGLAccount(@Body postFinancialActivityAccountsRequest: PostFinancialActivityAccountsRequest? = null): PostFinancialActivityAccountsResponse

    /**
     * Delete a Financial Activity to Account Mapping
     *
     * Responses:
     *  - 200: OK
     *
     * @param mappingId mappingId
     * @return [DeleteFinancialActivityAccountsResponse]
     */
    @DELETE("v1/financialactivityaccounts/{mappingId}")
    suspend fun deleteGLAccount(@Path("mappingId") mappingId: Long): DeleteFinancialActivityAccountsResponse

    /**
     * Retrieve a Financial Activity to Account Mapping
     * Example Requests:  financialactivityaccounts/1
     * Responses:
     *  - 200: OK
     *
     * @param mappingId mappingId
     * @return [GetFinancialActivityAccountsResponse]
     */
    @GET("v1/financialactivityaccounts/{mappingId}")
    suspend fun retreive(@Path("mappingId") mappingId: Long): GetFinancialActivityAccountsResponse

    /**
     * List Financial Activities to Accounts Mappings
     * Example Requests:  financialactivityaccounts
     * Responses:
     *  - 200: OK
     *
     * @return [kotlin.collections.List<GetFinancialActivityAccountsResponse]
     */
    @GET("v1/financialactivityaccounts")
    suspend fun retrieveAll(): List<GetFinancialActivityAccountsResponse>

    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @return [kotlin.String]
     */
    @GET("v1/financialactivityaccounts/template")
    suspend fun retrieveTemplate(): String

    /**
     * Update a Financial Activity to Account Mapping
     * the API updates the Ledger account linked to a Financial Activity
     * Responses:
     *  - 200: OK
     *
     * @param mappingId mappingId
     * @param postFinancialActivityAccountsRequest  (optional)
     * @return [PutFinancialActivityAccountsResponse]
     */
    @PUT("v1/financialactivityaccounts/{mappingId}")
    suspend fun updateGLAccount(
        @Path("mappingId") mappingId: Long,
        @Body postFinancialActivityAccountsRequest: PostFinancialActivityAccountsRequest? = null
    ): PutFinancialActivityAccountsResponse

}
